### ISSUE TITLE
Future challenge for PDF subset standards from Extensions

### ISSUE THREAD

**Issue by MPBailey** [pdf-association/pdf-issues#129]
Both PDF/X-6 and PDF/X-4 are based on 32000-2:2020, but neither make any comments about the Extensions dictionary. At present there are numerous extensions standards in development, many of which would make a file that uses them incompatible with readers that do not implement the extension.

Each of those extensions specifications requires that appropriate data be filled in the Extensions dictionary in a PDF file (32000-2:2020 7.12).

So we may soon find ourselves in a situation where a file is completely compliant with PDF/A-4 or PDF/X-6, but that cannot be read by the majority of readers for those formats.

On the up-side, at least those readers should be able to look in the Extensions dictionary and at least understand why they can’t read the file (and possibly provide that information to the user/operator).

It’s not clear what could/should be done about this now that the standards are published. I’m opening an issue here just to publicise the situation a bit more widely so that it comes as a surprise to fewer people when a valid file using a new image codec, or a non-rectangular link or whatever is encountered.

**Comment by MatthiasValvekens** [comment]
My two cents: I can see the argument for this being a potential source of problems in PDF/A-4 (given the focus of PDF/A on being a "future proof" subset). Even so: from the discussion on #100, I gather that even proprietary PDF features were never forbidden in PDF/A. While I don't have a particularly strong opinion on that issue, I do feel that dealing with explicitly declared extensions (or at least failing gracefully) is much _less_ of a hassle.

In other words: if private/undocumented features are OK in PDF/A, then surely "Annex E-compliant" extensions are fine too, right?

I can't comment on PDF/X-6 since that's too far out of my area of expertise.

**Comment by lrosenthol** [comment]
When we talked about extensions in the past, we were only considering ones that did not impact rendering - so non-rectangular links would be just fine.  

However, with the additions for 3D and encryption (supported in profiles of X-4 and in X-6), not to mention new image codes or the >4 color blending space work that has been proposed - we are now looking at extensions that go beyond where we considered.

Agreed this is a good topic and worth discussion at 171/WG 5 and 130/WG 2

**Comment by petervwyatt** [comment]
This issue was discussed during the ISO TC 171 SC 2 WG 5 May 2022 Meeting. 

The decision from WG5 was to ask the PDF Association for a guidance report on how the aligned PDF subset standards (i.e. PDF/A and PDF/X) should interpret and understand their specifications when such PDF files include ISO extensions, private extensions, private data, second class names, etc. This report will be submitted to the Fall 2022 WG5 meeting and the next ISO TC 130 WG 2 meeting.

**Comment by lrosenthol** [comment]
This has been resolved as part of the X/6<->A/4 alignment

**Comment by petervwyatt** [comment]
Great!. 
Labelling as "proposed solution" so it is on my radar for the next PDF TWG meeting and can discuss and hopefully close 'no-fix' (here) as solved by the dated revisions of PDF/X and PDF/A.

**Comment by MPBailey** [comment]
Resolving in "Conforming files" clause of X-6 by prohibition of features in the file that affect "final print reproduction"; equivalent in A-4.

**Comment by lrosenthol** [comment]
Additional comment from @MPBailey 

>As I said on the A vs X call today, one of the things that misleads developers most when reading standards is the tendency to extrapolate from one requirement to something else. I've spent a lot of time training people that each statement should be read exactly as written, and does not imply anything beyond that range. As I said, my example is a standard that says "if A is true, then do B" does not say anything about what you should do if A is not true; you'll have to look elsewhere in the doc to find out what to do then.
>That's why I think "A PDF/X-6 conforming file may also include other valid PDF features that do not affect final print reproduction of the compound entity" is not sufficiently clear in prohibiting features that do or might affect reproduction, even with the note that was added on today's call.
>I'd be much happier with something like "A PDF/X-6 conforming file shall not include other PDF features that affect final print reproduction of the  compound entity", and perhaps tweak the new note to say that valid extensions that do not affect the print reproduction can be used.
>To make it suitable for both X and A perhaps use "rendered appearance" rather than "print reproduction".

**Comment by DietrichSeggern** [comment]
I do not think that we should introduce something like "rendered appearance" into PDF/A. The provision in PDF/X relates in my opinion to things that have an impact on the appearance as rendered on screen but not on print. There is no comparable thing in PDF/A.

**Comment by petervwyatt** [comment]
PDF TWG agree that this is handled by the "Aligning PDF/A and PDF/X" LWG with dated revisions of both those standards. No fix needed in ISO 32000-2.
